snapcraft/meta/appstream.py

```diff
@@ -186,12 +187,15 @@ def _get_value_from_xml_element(tree, key) -> str | None:
     return None
 
 
-def _get_urls_from_xml_element(nodes, url_type) -> list[str] | None:
+def _get_urls_from_xml_element(
+    nodes: list[lxml.etree._Element], url_type: str
+) -> list[str] | None:
     urls: list[str] = []
     for node in nodes:
         if (
             node is not None
             and node.attrib["type"] == url_type
+            and node.text is not None
```

Comment:
This change does not seem related to typing. Is this a fix that was spotted when the typing was fixed?

Reply:
What happened is that once `nodes` was typed, the type checkers were deduced that `node.text` is an optional field.

That caused the type-checkers to complain about the next line `node.text.strip()` on a `str | None`.

So this is a bug that was exposed due to improved type checking.  I think it's a safe fix, so I made the change.

Reply:
Oh nice! Thanks for the explanation.